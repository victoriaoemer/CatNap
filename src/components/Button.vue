<script setup lang="ts" name="Button">
defineProps({
  text: String,
});

</script>

<template>
  <button class="px-5 py-2  gradient-border">{{ text }}</button>
</template>
