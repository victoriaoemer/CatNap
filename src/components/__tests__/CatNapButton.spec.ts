import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import Button from '@/components/CatNapButton.vue';

describe('CatNapButton.vue', () => {
  it('renders a filled button with the correct text', () => {
    const wrapper = mount(Button, {
      props: {
        text: 'Click Me',
        type: 'filled',
      },
    });

    // controls if the filled button is rendered
    const button = wrapper.find('button.gradient-bg');
    expect(button.exists()).toBe(true);

    // controls if the text is rendered correctly
    expect(button.text()).toBe('Click Me');
  });

  it('renders an outline button with the correct text', () => {
    const wrapper = mount(Button, {
      props: {
        text: 'Outline Button',
        type: 'outline',
      },
    });

    // controls if the outline button is rendered
    const button = wrapper.find('button.border');
    expect(button.exists()).toBe(true);

    // controls if the text is rendered correctly
    expect(button.text()).toBe('Outline Button');
  });

  it('does not render any button if type is invalid', () => {
    const wrapper = mount(Button, {
      props: {
        text: 'Invalid Button',
        type: 'invalid',
      },
    });

    // controls if the button is not rendered
    expect(wrapper.find('button').exists()).toBe(false);
  });

  // snapshot testing
  it('matches snapshot for filled button', () => {
    const wrapper = mount(Button, {
      props: {
        text: 'Snapshot Button',
        type: 'filled',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches snapshot for outline button', () => {
    const wrapper = mount(Button, {
      props: {
        text: 'Snapshot Button',
        type: 'outline',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
