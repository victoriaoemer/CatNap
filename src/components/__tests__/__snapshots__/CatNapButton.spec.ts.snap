// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CatNapButton.vue > matches snapshot for filled button 1`] = `
"<div data-v-110b83fa=""><button data-v-110b83fa="" class="gradient-bg hover:bg-gradient">Snapshot Button</button>
  <!--v-if-->
</div>"
`;

exports[`CatNapButton.vue > matches snapshot for outline button 1`] = `
"<div data-v-110b83fa="">
  <!--v-if--><button data-v-110b83fa="" class="border rounded-xl border-secondary hover:bg-gradient">Snapshot Button</button>
</div>"
`;
